grammar.js

```diff
@@ -506,8 +506,9 @@ module.exports = grammar({
 		_semis: $ => /[\r\n]+/,
 
 		assignment: $ => choice(
-			prec.left(PREC.ASSIGNMENT, seq($.directly_assignable_expression, $._assignment_and_operator, $._expression)),
-			prec.left(PREC.ASSIGNMENT, seq($.directly_assignable_expression, "=", $._expression))
+			prec.left(PREC.ASSIGNMENT, seq($._primary_expression, $._assignment_and_operator, $._expression)),
+      prec.left(PREC.ASSIGNMENT, seq($._primary_expression, "=", $._expression)),
+      prec.left(PREC.ASSIGNMENT, seq($._primary_expression, $.navigation_suffix, "=", $._expression))
```

Comment:
This is nice, and it might simplify the original grammar of Kotlin, but I'm a bit worried that we deviate now even
more from the original grammar. Fwcd already deviated from the original grammar a little on how he is handling expressions. I'd rather see us go back to the original spec of Kotlin where there is a directly_assignable_expression,
and a postfix_unary_operator, etc.

Reply:
I see, good point! I thought that it would be nice to simplify it, but you're right in the fact that deviating more might make later work harder. i'll look into it some more and see if I can make it work by sticking closer to the original grammar 👍